tests/fixtures/new_fixtures.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+"""These fixtures are shared globally across the test suite."""
+
+from unittest.mock import Mock
+
+import pytest
+
+from composer.core import Logger, State
+from composer.core.types import Precision
+
+
+@pytest.fixture
+def minimal_state():
```

Comment:
I would ideally like the`minimal_state` be something that behaves as similarly to the real state, so that way tests only need to set what is needed, and not worry about replacing the mock parameters. Having partially defined mocks in global fixtures hide a pretty significant difference in the fixture construction, leading to harder-to-debug tests.

As an alternative of having just one `minimal_state`, could we have one state fixture for each model type? E.g. `simple_model_state` and `simple_conv_model_state`? Each state could then have a dataset of the appropriate type.
Then, could we update `State.__init__` to default `grad_accum`  to 1, precision to FP32, and evaluators to None or an empty collection (to match the Trainer's init signature)?

I think these changes combined would allow us to remove the mocks from the fixtures while ensuring that it would be something trainable.

Reply:
Great feedback, thanks! I did all of the above, but instead of having `simple_model_state` and `simple_conv_model_state`, just made `minimal_state == simple_model_state`. We can introduce a `conv_model_state` if the tests need that many times, but right now its more niche applications that require a little tinkering to the dataset configurations, etc.